### ISSUE TITLE
Broken Link:https://index.ros.org/doc/ros2/Tutorials/Using-Python-Packages/#installing-via-a-virtual-environment

### ISSUE THREAD

**Issue by ShashwatPatil** [ros-infrastructure/rosindex#389]
### 404 Page URL

https://index.ros.org/doc/ros2/Tutorials/Using-Python-Packages/#installing-via-a-virtual-environment

### Referrer URL

https://answers.ros.org/question/371083/how-to-use-python-virtual-environments-with-ros2/

### Description

the link is broken, i am tryinng to use ros on a python virtual env.... pls help

**Comment by CursedRock17** [comment]
Here's the tutorial you're looking for on the [rolling](https://docs.ros.org/en/rolling/How-To-Guides/Using-Python-Packages.html) distribution, that being said, if you want stability go to the [jazzy](https://docs.ros.org/en/jazzy/How-To-Guides/Using-Python-Packages.html) distribution. I've added the correct link to the [correct forum](https://robotics.stackexchange.com/questions/98214/how-to-use-python-virtual-environments-with-ros2) in robotics.stackexchange.com.

**Comment by ShashwatPatil** [comment]
Thanks for the solution, but i actually am working on cbfkit which uses ros2 humble and python lib jax and hence requires python 3.10
Also the other pkgs that i am using CBFKit on uses ros foxy so there has been a conflicts on finding rclpy, although foxy has it installed 

I also ran this toolbox(cbfkit) on a docker image with ros humble and it is working fine, but being on docker i am not able to utilize the cuda cores.

I also tried source installation of humble on my ubuntu 20.04 but the structure of source installed ros is quite different that normal installation one.

Thanks again for the solution though... Will try that and see if it works.
